tests/test_name_resolution/good/toplevel_defs.catala_en

```diff
@@ -192,3 +191,284 @@ let S_9 (S_in_32: S_in {}) =
+            return False
+
+    def __ne__(self, other: object) -> bool:
+        return not (self == other)
+
+    def __str__(self) -> str:
+        return "S3(a={})".format(self.a)
+
+class S4:
+    def __init__(self, a: Decimal) -> None:
+        self.a = a
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, S4):
+            return (self.a == other.a)
+        else:
+            return False
+
+    def __ne__(self, other: object) -> bool:
+        return not (self == other)
+
+    def __str__(self) -> str:
+        return "S4(a={})".format(self.a)
+
+class S:
+    def __init__(self, a: Decimal, b: A) -> None:
+        self.a = a
+        self.b = b
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, S):
+            return (self.a == other.a and self.b == other.b)
+        else:
+            return False
+
+    def __ne__(self, other: object) -> bool:
+        return not (self == other)
+
+    def __str__(self) -> str:
+        return "S(a={},b={})".format(self.a, self.b)
+
+class SIn:
+    def __init__(self, ) -> None:
+        pass
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, SIn):
+            return (True)
+        else:
+            return False
+
+    def __ne__(self, other: object) -> bool:
+        return not (self == other)
+
+    def __str__(self) -> str:
+        return "SIn()".format()
+
+class S2In:
+    def __init__(self, ) -> None:
+        pass
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, S2In):
+            return (True)
+        else:
+            return False
+
+    def __ne__(self, other: object) -> bool:
+        return not (self == other)
+
+    def __str__(self) -> str:
+        return "S2In()".format()
+
+class S3In:
+    def __init__(self, ) -> None:
+        pass
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, S3In):
+            return (True)
+        else:
+            return False
+
+    def __ne__(self, other: object) -> bool:
+        return not (self == other)
+
+    def __str__(self) -> str:
+        return "S3In()".format()
+
+class S4In:
+    def __init__(self, ) -> None:
+        pass
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, S4In):
+            return (True)
+        else:
+            return False
+
+    def __ne__(self, other: object) -> bool:
+        return not (self == other)
+
+    def __str__(self) -> str:
+        return "S4In()".format()
+
+
+
+glob1 = (decimal_of_string("44.12"))
+
+def glob3(x:Money):
+    return (decimal_of_money(x) + decimal_of_string("10."))
+
+def glob4(x_1:Money, y:Decimal):
+    return ((decimal_of_money(x_1) * y) + decimal_of_string("10."))
+
+def glob5_aux():
+    def glob5(x_2:Decimal):
+        y_1 = decimal_of_string("1000.")
+        return (x_2 * y_1)
+    return glob5((decimal_of_integer(integer_of_string("2")) *
+                 decimal_of_string("3.")))
+
+glob5_1 = (glob5_aux())
```

Comment:
Is this the wanted behavior? Why pass through this auxiliary variable ?

Reply:
I have been conservative in the compilation to `scalc`, but believe it can be simplified indeed (I've started some work in that direction, will be happy to discuss it at the next occasion), here and in a few other cases.

The scheme I've chosen at the moment, whenever the computation of a toplevel variable spills out statements, is to wrap its computation in an auxiliary function with no arguments, and apply that just once. The output seems consistent with that, but it could probably be simplified and flattened.